webpack/assets/javascripts/react_app/components/PF4/InlineEdit/InlineEdit.js

```diff
@@ -0,0 +1,136 @@
+import React, { useState, useEffect } from 'react';
+import {
+  Spinner,
+  TextInput,
+  TextArea,
+  Text,
+  TextVariants,
+  Button,
+  Split,
+  SplitItem,
+} from '@patternfly/react-core';
+import { TimesIcon, CheckIcon, PencilAltIcon } from '@patternfly/react-icons';
+import PropTypes from 'prop-types';
+import { translate as __ } from '../../../common/I18n';
+import './inlineEdit.scss';
+
+const InlineEdit = ({ onEdit, value, textArea, attribute }) => {
+  // Tracks input box state
+  const [inputValue, setInputValue] = useState(value);
+  const [editing, setEditing] = useState(false);
+  const [submitting, setSubmitting] = useState(false);
+
+  // Setting didCancel to avoid actions from happening after component has been unmounted
+  // see https://overreacted.io/a-complete-guide-to-useeffect/#speaking-of-race-conditions
+  useEffect(() => {
+    let didCancel = false;
+
+    const onSubmit = async () => {
+      if (submitting) {
+        await onEdit(inputValue, attribute);
+        if (!didCancel) {
+          setSubmitting(false);
+          setEditing(false);
+        }
+      }
+    };
+    onSubmit();
+
+    return () => {
+      didCancel = true;
+    };
+  }, [submitting]);
+
+  // Listen for enter and trigger submit workflow on enter
+  useEffect(() => {
+    const listener = event => {
+      if (event.code === 'Enter' || event.code === 'NumpadEnter') {
+        if (editing) setSubmitting(true);
+      }
+    };
+    document.addEventListener('keydown', listener);
```

Comment:
Adding the event listener to the `document` can cause problems:  If you have more than one `<InlineEdit>` on the page, you can get into a state where you are `editing` both, and pressing Enter will submit both.  I verified this by copy-pasting the component in the storybook.  Worse: the value of one input overrode the other, and they both ended up with the same value.

Adding the event listener to the actual DOM element would avoid the issue, but then you'd need a ref.  Both the `<TextArea>` and `<TextInput>` PF components do offer an `innerRef` prop that gets you access to the underlying `<input>` element, if you're inclined to explore that.

Reply:
also of note: [`<TextInput>`](https://patternfly-react.surge.sh/components/text-input#props) offers an `onBlur` callback, not sure if that would get called on Enter but worth a shot

Reply:
playing around with this, it seems `innerRef` doesn't work at all.  (not sure if it's a PF bug or what)  But just using `ref` actually gives you access to the underlying `input` element because PF is using `React.forwardRef`.  So the new `ref` approach is correct.